cmake/util/FindCUDA.cmake

```diff
@@ -37,7 +37,7 @@
 #
 macro(find_cuda use_cuda)
   set(__use_cuda ${use_cuda})
-  if(__use_cuda STREQUAL "ON")
+  if(${__use_cuda})
```

Comment:
we should call decompose_arg on this case as well. Most of the command will require decompose_arg

Reply:
This case does not need any special treatment, it is checking if __use_cuda is a boolean true value. If __use_cuda is a string (path) then this test will fail and the test on line 42 below is checked.

Reply:
For readability, let us do `${USE_CUDA} MATCHES ${IS_TRUE_PATTERN}` :)